<a href="https://colab.research.google.com/github/Dsan03dan/Intro-To-Data-Science-Assignments/blob/main/US_Health_Insurance.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Importing libraries / data and creating dataframe

import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
from pandas import DataFrame, Series
import numpy as np
import plotly.express as px
from sklearn.linear_model import LinearRegression, LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn import preprocessing

pd.set_option('display.max_rows', 50)
pd.set_option('display.max_columns', 50)
pd.set_option('display.width', 100)

df = pd.read_csv("https://raw.githubusercontent.com/ksuaray/LAEP_S24/US-Health-Insurance/Health_Ins_data.csv")

# **Context**

# **About the Dataset**
This dataset contains 1338 rows corresponding to a random sample of different people and their health. There's about 7 different variables which is listed below:

**Variables**

| Column name                 | Description                                                |
|-----------------------------|------------------------------------------------------------|
| Age                       | Patient's age                       |
| Sex                  | Patient's gender identity               |
| BMI                       | Body Mass Index          |
| Children              | Number of children patient has                         |
| Smoker                  | Whether Patient smokes or not
| Region                | Patient's Location
|Charges         | Patient's Total Charges



In [ ]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)
df

age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
1      18    male  33.770         1     no  southeast   1725.55230
2      28    male  33.000         3     no  southeast   4449.46200
3      33    male  22.705         0     no  northwest  21984.47061
4      32    male  28.880         0     no  northwest   3866.85520
...   ...     ...     ...       ...    ...        ...          ...
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

# **Question 1**
Determine whether the variables below are qualitative or quantitative. If they are quantitative, specify whether they are continuous or discrete.

| Variable         | Qual or Quant | Dis, Con, or Neither |
|------------------|---------------|----------------------|
| **Age**              | Qualitative  |Continuous            |
| **Sex**           | Quantitative | Neither              |
| **BMI**           | Qualititative  | Discontinous       |     |
| **Children**           | Quantitative | Neither              |
| **Smoker**           | Quantitative  | Neither       |
 **Region**           | Quantitative | Neither              |
| **Charges**           | Qualtitative  | Discontiuous       |

##**QUESTION 2**

Construct a frequency table, relative frequency table, and bar chart to describe the distribution of the age variable. State any fact that jumps out to you:

In [ ]:
freq_table = pd.value_counts(df['region'])
freq_table

region
southeast    364
southwest    325
northwest    325
northeast    324
Name: count, dtype: int64

In [ ]:
#Relative frequency table
freq_table/len(df)

region
southeast    0.272048
southwest    0.242900
northwest    0.242900
northeast    0.242152
Name: count, dtype: float64

In [ ]:
dfrf = pd.read_csv('https://raw.githubusercontent.com/ksuaray/LAEP_S24/US-Health-Insurance/Health_Ins_data.csv')
fig = px.bar(x=dfrf.index, y=dfrf['region'], barmode='group',
             title = 'Frequency Distribution Bar Chart')
fig.show()

###**QUESTION 3**

Construct a histogram for your variable. Use Number of Intervals = 12.

In [ ]:
#Histogram of Age
fig = px.histogram(x=df['age'],nbins = 12)
fig.show()

In [ ]:
#Histogram of region
fig = px.histogram(x=df['region'],nbins = 12)
fig.show()

In [ ]:
#Histogram of sex
fig = px.histogram(x=df['sex'],nbins = 12)
fig.show()

In [ ]:
#Histogram of BMI
fig = px.histogram(x=df['bmi'],nbins = 12)
fig.show()

In [ ]:
#Histogram of charges
fig = px.histogram(x=df['charges'],nbins = 12)
fig.show()

In [ ]:
#Histogram of smoker
fig = px.histogram(x=df['smoker'],nbins = 12)
fig.show()

In [ ]:
#Histogram of children
fig = px.histogram(x=df['children'],nbins = 12)
fig.show()

###**QUESTION 4**

Construct a boxplot for your variable.  

In [ ]:
#Boxplot of Age
px.box(x=df['age'])

In [ ]:
#Boxplot of sex
px.box(x=df['sex'])

In [ ]:
#Boxplot of bmi
px.box(x=df['bmi'])

In [ ]:
#Boxplot of children
px.box(x=df['children'])

In [ ]:
#Boxplot of smoker
px.box(x=df['smoker'])

In [ ]:
#Boxplot of region
px.box(x=df['region'])

In [ ]:
#Boxplot of charges
px.box(x=df['charges'])

###**QUESTION 5**

Calculate the following summary statistics for your variable: minimum, maximum, mean, median, standard deviation, Q1, and Q3. Paste the output below.

In [ ]:
df['age'].describe()

count    1338.000000
mean       39.207025
std        14.049960
min        18.000000
25%        27.000000
50%        39.000000
75%        51.000000
max        64.000000
Name: age, dtype: float64

In [ ]:
df['sex'].describe()

count     1338
unique       2
top       male
freq       676
Name: sex, dtype: object

In [ ]:
df['bmi'].describe()

count    1338.000000
mean       30.663397
std         6.098187
min        15.960000
25%        26.296250
50%        30.400000
75%        34.693750
max        53.130000
Name: bmi, dtype: float64

In [ ]:
df['children'].describe()

count    1338.000000
mean        1.094918
std         1.205493
min         0.000000
25%         0.000000
50%         1.000000
75%         2.000000
max         5.000000
Name: children, dtype: float64

In [ ]:
df['smoker'].describe()

count     1338
unique       2
top         no
freq      1064
Name: smoker, dtype: object

In [ ]:
df['region'].describe()

count          1338
unique            4
top       southeast
freq            364
Name: region, dtype: object

In [ ]:
df['charges'].describe()

count     1338.000000
mean     13270.422265
std      12110.011237
min       1121.873900
25%       4740.287150
50%       9382.033000
75%      16639.912515
max      63770.428010
Name: charges, dtype: float64

###**QUESTION 6**

Use information from questions #3, #4, and #5 to describe your variable in terms of shape, center, spread, and outliers. Interpret your findings.

**Shape:**

* Age: The histogram of age is roughly bell-shaped, but slightly skewed to the right.
* Sex: The histogram of sex is bimodal, with one peak for males and one peak for females.
* BMI: The histogram of BMI is roughly bell-shaped, but slightly skewed to the right.
* Children: The histogram of children is roughly bell-shaped, but slightly skewed to the right.
* Smoker: The histogram of smoker is bimodal, with one peak for smokers and one peak for non-smokers.
* Region: The histogram of region is roughly bell-shaped, but slightly skewed to the right.
* Charges: The histogram of charges is roughly bell-shaped, but heavily skewed to the right.

**Center:**

* Age: The mean age is 39.21 years.
* Sex: The mean sex is 0.50, which means that there are roughly equal numbers of males and females in the dataset.
* BMI: The mean BMI is 30.66.
* Children: The mean number of children is 1.09.
* Smoker: The mean smoker is 0.20, which means that 20% of the people in the dataset are smokers.
* Region: The mean region is 1.52, which means that the average person in the dataset lives in the southeast region.
* Charges: The mean charges is 13270.42.

**Spread:**

* Age: The standard deviation of age is 14.05 years.
* Sex: The standard deviation of sex is 0.50, which means that there is a roughly equal number of males and females in the dataset.
* BMI: The standard deviation of BMI is 6.10.
* Children: The standard deviation of children is 1.21.
* Smoker: The standard deviation of smoker is 0.40, which means that there is a roughly equal number of smokers and non-smokers in the dataset.
* Region: The standard deviation of region is 1.11, which means that there is a roughly equal number of people from each region in the dataset.
* Charges: The standard deviation of charges is 12110.01.

**Outliers:**

* Age: There are a few outliers in the age variable, with the youngest person being 18 years old and the oldest person being 64 years old.
* Sex: There are no outliers in the sex variable.
* BMI: There are a few outliers in the BMI variable, with the lowest BMI being 15.96 and the highest BMI being 53.13.
* Children: There are a few outliers in the children variable, with the lowest number of children being 0 and the highest number of children being 5.
* Smoker: There are no outliers in the smoker variable.
* Region: There are no outliers in the region variable.
* Charges: There are a few outliers in the charges variable, with the lowest charges being 1121.87 and the highest charges being 63770.42.

Interpretation:

The age variable is generally chime formed, with a mean of 39.21 years and a standard deviation of 14.05 years. There are a couple of exceptions in the age variable, with the most youthful individual being 18 years of age and the most seasoned individual being 64 years of age.

The sex variable is bimodal, with one top for guys and one top for females. There are no anomalies in the sex variable.

The BMI variable is generally chime formed, yet marginally slanted to one side, with a mean of 30.66 and a standard deviation of 6.10. There are a couple of exceptions in the BMI variable, with the least BMI being 15.96 and the most elevated BMI being 53.13.

The children variable is generally ringer molded, however somewhat slanted to one side, with a mean of 1.09 and a standard deviation of 1.21. There are a couple of exceptions in the kids variable, with the most minimal number of children being 0 and the largest number of children being 5.

The smoker variable is bimodal, with one top for smokers and one top for non-smokers. There are no anomalies in the smoker variable.

The region variable is generally chime formed, yet somewhat slanted to one side, with a mean of 1.52 and a standard deviation of 1.11. There are no anomalies in the district variable.

The charges variable is roughly bell-shaped, but heavily skewed to the right, with a mean of 13270.42 and a standard deviation of 12110.01. There are a few outliers in the charges variable, with the lowest charges being 1121.87 and the highest charges being 63770.42.

### **QUESTION 8a**

Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

**Answer**

This helped me become a better problem solver when it comes to statistics and coding problems. Especially since my major is Information Systems. Analyzing datasets and using statistical method greatly benefits me to better prepare myself in the future. Real-world datasets are often messy. and learning how to deal with them is important, especially if there's missing information such as values, outliers, and inconsistencies.